### PR TITLE
Convenience methods and housekeeping

### PR DESCRIPTION
**Pull request by bkonyi**
Cleanup changes:  
- Migrated most tests away from using the property method by adding
convenience accessors.
- Changed most import paths from using relative paths to fully resolved
  package:draw/ paths

### CODE REVIEW

File: lib/src/models/comment_impl.dart [hunk 1]
```diff
@@ -136,77 +139,81 @@ class MoreComments extends RedditBase {
...
-  CommentForest _replies;
+class Comment extends CommentRef {
+  // CommentModeration get mod; // TODO(bkonyi): implement
 
   /// Returns true if the current [Comment] is a top-level comment. A [Comment]
   /// is a top-level comment if its parent is a [Submission].
-  Future<bool> get isRoot async {
-    final parentIdType = (await _parentId).split('_')[0];
+  bool get isRoot {
+    final parentIdType = parentId.split('_')[0];
```
Comment (ckartik) [inline lib/src/models/comment_impl.dart]: Looked into the docs and it says that this is the goto strategy:  "you might call an async function from your function".

Guessing that's what you're doing here but how come you don't have to handle the parentId as a Future type?

File: lib/src/models/comment_impl.dart [hunk 2]
```diff
@@ -8,16 +8,16 @@ import 'dart:async';
 // Required for `hash2`
 import 'package:quiver/core.dart';
```
Comment (ckartik) [inline lib/src/models/comment_impl.dart]: What's the purpose of using a private implementation for the Comment object?

File: lib/src/models/comment_impl.dart [hunk 3]
```diff
@@ -136,77 +139,81 @@ class MoreComments extends RedditBase {
...
-      : super.withPath(reddit, _infoPath(id));
-
-  static String _infoPath(String id) =>
-      apiPath['comment'].replaceAll(_commentRegExp, id);
-
-  Future<String> get _parentId async {
-    return await property('parentId');
-  }
+  /// The ID of the parent [Comment] or [Submission].
+  String get parentId => data['parent_id'];
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply lib/src/models/comment_impl.dart]: I think you're right. I'll take a look.

[the remark this answers is not in the document]
Comment (bkonyi) [reply lib/src/models/comment_impl.dart]: The second getter is part of `MoreComments`, but that shouldn't have been a `Future` either way.

File: lib/src/models/comment_impl.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (bkonyi) [reply lib/src/models/comment_impl.dart]: We have some methods that are accessing internal fields of `Comment` and `Submission` objects, so this is just a way to hide these methods. See `comment.dart` and `submission.dart` to see what I mean.

File: lib/src/models/comment_impl.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (bkonyi) [reply lib/src/models/comment_impl.dart]: Not sure what you mean by this.

**Pull request merged**